### PR TITLE
remove OSSL_HTTP ->OCSP renaming

### PR DESCRIPTION
**Pull request by richsalz**
This is built on #13620 but could probably be split.

The way things were renamed to look like ```HTTP_OCSP_xxx``` as opposed to ```OCSP_xxx``` is wrong.  The only thing enabled was that the implementation in crypto/http/http_client.c had "new" names, but a set of defines in the internal local header file crypto/http/http_local.h mapped those new names to "old" style names. The "old" names is what were exported.

I spent more than an hour trying to find definitions of documented functions, etc., but failed because of the "new" names.

### CODE REVIEW

**Comment by richsalz** [comment]
Also, a comment "this function isn't used" was wrong.

File: crypto/http/http_local.h [hunk 1]
```diff
@@ -15,17 +15,6 @@
 
 /* name aliases for legacy names with name prefix "OCSP_" */
 typedef OCSP_REQ_CTX OSSL_HTTP_REQ_CTX;
```
Comment (levitte) [inline crypto/http/http_local.h:17]: Is there a reason you're leaving this typedef alone? Following your argument, it should be just as wrong as the function aliases you removed.

Comment (richsalz) [reply crypto/http/http_local.h:17]: Yeah, I thought about that and probably made the wrong discussion.  Additional commit coming up.

**Comment by richsalz** [comment]
CI failure is the encode/decode thing and not relevant.

**Review by DDvO — CHANGES_REQUESTED** [review]
The 'solution' taken here to revert the name changes I did from `OCSP_` to `OSSL_HTTP_` back to `OCSP` is IMO entirely wrong!

Didn't you see that the data structure and the functions have been generalized from an OCSP-biased bare-bones HTTP client to a generic one where OCSP is just an instance?

I deliberately had kept the old names internally to simplify backward compatibility, though I admit that this can be confusing. Maybe a better way of achieving that would have been to rename the original function and provide #define's for providing access via the old names.

**Comment by richsalz** [comment]
Sorry @ddvo I strongly disagree.  Renaming the *implementation but nothing else* is really wrong IMO.

But if the project rejects this, that's okay.  I don't know the mechanism if contributors can put a hold, but perhaps some group in the project (otc or committers) wishes to discuss it.

**Comment by DDvO** [comment]
> Sorry @DDvO I strongly disagree. Renaming the _implementation but nothing else_ is really wrong IMO.

@richsalz, have you understood what #11404 was all about?
Your response did not sound like that.
Note that both the data structure and the function were not just renamed, but strongly generalized.
The old OCSP-focused client is history.

**Comment by DDvO** [comment]
Oops, I meant the HTTP client as introduced PR #10667.
Renaming the functionality back to 'OCSP' would not make sense.

**Comment by richsalz** [comment]
You are right, I didn't realize that ```OSSL_HTTP``` was documented and exposed. So the way the new names were handled is wrong. Will you fix it, or should I?  I will close this PR.

**Comment by DDvO** [comment]
All right. Let's first clarify how to solve the confusion the right way. How about the idea I gave above?
At the example of `#define OSSL_HTTP_REQ_CTX_new OCSP_REQ_CTX_new` this would mean:
* rename the old/original function `OCSP_REQ_CTX_new` to `OSSL_HTTP_REQ_CTX_new`
* for backward compatibility, `#define OCSP_REQ_CTX_new OSSL_HTTP_REQ_CTX_new
* state in the doc that `OCSP_REQ_CTX_new` is deprecated and superseded by `OSSL_HTTP_REQ_CTX_new`

If you agree, would be nice if you can take over this, which could be done even as new version of this PR.

**Comment by richsalz** [comment]
I agree and will do the deprecation as you describe.

**Comment by richsalz** [comment]
Closing in favor of #13742.

**Pull request closed without merging**